File: src/tests/DataBlockASPNETDemo/csharp/Mappings/Author.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace BookManager.Mappings
{

    [Serializable()]
    public class Author : TableMetadata
    {
        private DatabaseField[] _fields;

        public Author()
        {
            _fields = new DatabaseField[4];
            _fields[0] = new DatabaseField(DbType.Int32, "AuthorId", true, true, null);
            _fields[1] = new DatabaseField(DbType.String, "Name", false, false, null);
            _fields[2] = new DatabaseField(DbType.Int32, "Age", false, false, null);
            _fields[3] = new DatabaseField(DbType.String, "Location", false, false, null);

            this.currentTableName = "Author";

            this.listRelations.Add(new ParentTableRelation("BookAuthors", "AuthorId", TableRelationCardinality.OneToMany, true));
            this.listRelations.Add(new ManyToManyTableRelation("Book", "BookAuthors", "AuthorId", "BookId"));

        }


        public override DatabaseField[] TableFields
        {
            get
            {
                return _fields;
            }
            set
            {
                _fields = value;
            }
        }

        public System.Int32 AuthorId
        {
            get
            {
                return (System.Int32)(this.GetField("AuthorId")).fieldValue;
            }

            set
            {
                this.SetFieldValue("AuthorId", value);
            }
        }


        public System.String Name
        {
            get
            {
                return (System.String)(this.GetField("Name")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Name", value);
            }
        }


        public System.Int32 Age
        {
            get
            {
                return (System.Int32)(this.GetField("Age")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Age", value);
            }
        }


        public System.String Location
        {
            get
            {
                return (System.String)(this.GetField("Location")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Location", value);
            }
        }

        public Array GetBookAuthors()
        {
            Type relatedType = Type.GetType("BookManager.Mappings.BookAuthors");
            return this.GetRelatedTableData(relatedType);
        }
        public Array GetBook()
        {
            Type relatedType = Type.GetType("BookManager.Mappings.Book");
            Type intermediaryType = Type.GetType("BookManager.Mappings.BookAuthors");
            return this.GetRelatedTableData(relatedType, intermediaryType);
        }
    }
}

File: src/tests/DataBlockASPNETDemo/vb.net/mappings/Author.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace BookManager.Mappings

	<Serializable()> _
	Public Class Author
	Inherits TableMetadata
		Private _fields As DatabaseField()

		Public Sub New()
            _fields = New DatabaseField(3) {}
			_fields(0) = New DatabaseField(DbType.Int32, "AuthorId", True, True, Nothing)
			_fields(1) = New DatabaseField(DbType.String, "Name", False, False, Nothing)
			_fields(2) = New DatabaseField(DbType.Int32, "Age", False, False, Nothing)
			_fields(3) = New DatabaseField(DbType.String, "Location", False, False, Nothing)
			Me.currentTableName = "Author"
			Me.listRelations.Add(New ParentTableRelation("BookAuthors", "AuthorId", TableRelationCardinality.OneToMany, True))
			Me.listRelations.Add(New ManyToManyTableRelation("Book", "BookAuthors", "AuthorId", "BookId"))
		End Sub

		Public Overloads Overrides Property TableFields() As DatabaseField()
			Get
				Return _fields
			End Get
			Set
				_fields = value
			End Set
		End Property

		Public Property AuthorId() As System.Int32
			Get
				Return CType((Me.GetField("AuthorId")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("AuthorId", value)
			End Set
		End Property

		Public Property Name() As System.String
			Get
				Return CType((Me.GetField("Name")).fieldValue, System.String)
			End Get
			Set
				Me.SetFieldValue("Name", value)
			End Set
		End Property

		Public Property Age() As System.Int32
			Get
				Return CType((Me.GetField("Age")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("Age", value)
			End Set
		End Property

		Public Property Location() As System.String
			Get
				Return CType((Me.GetField("Location")).fieldValue, System.String)
			End Get
			Set
				Me.SetFieldValue("Location", value)
			End Set
		End Property

		Public Function GetBookAuthors() As Array
			Dim relatedType As Type = Type.GetType("BookManager.Mappings.BookAuthors")
			Return Me.GetRelatedTableData(relatedType)
		End Function

		Public Function GetBook() As Array
			Dim relatedType As Type = Type.GetType("BookManager.Mappings.Book")
			Dim intermediaryType As Type = Type.GetType("BookManager.Mappings.BookAuthors")
			Return Me.GetRelatedTableData(relatedType, intermediaryType)
		End Function
	End Class 
End Namespace
